### ISSUE TITLE
Using "root" and "nodir" at the same time includes directories in your results

### ISSUE THREAD

**Issue by lostthetrail** [isaacs/node-glob#221]
As stated in the title, if you use root and nodir options, directories are included in your results.

**Comment by lostthetrail** [comment]
```
t.test('a nodir', function (t) {
  var g = glob('/b*/**', { root: path.resolve('a'), nodir: true }, function (er, matches) {
    t.ifError(er)
    var wanted = [
        '/b/c/d', '/bc/e/f'
      ].map(function (m) {
        return path.join(path.resolve('a'), m).replace(/\\/g, '/')
      })

    t.like(matches, wanted)
    cacheCheck(g, t)
    t.end()
  })
})
```

**Comment by unional** [comment]
I have the same problem with cwd + nodir.

**Comment by isaacs** [comment]
Fixed on 7.0.2.  Thanks for finding the bug!

**Comment by mascardoso** [comment]
```
glob('**/*.png', {
  cwd: path.resolve(__dirname, 'assets/img'),
  nodir: true
}, (er, files) => {
  console.log(files)
})
```

^ this still returns `dir/filename.png` instead of just `filename.png`
